core/modules/layout/layout.admin.inc

```diff
@@ -328,20 +328,21 @@ function layout_settings_form($form, &$form_state, Layout $layout) {
     '#parents' => array('context'),
   );
 
-  $form['context_wrapper']['context']['add'] = array(
-    '#type' => 'submit',
-    '#value' => t('Add context'),
-    '#attributes' => array('class' => array('layout-link-button', 'layout-access-add')),
-    '#validate' => array(),
-    '#submit' => array(
-      'layout_settings_form_update_layout',
-      'layout_settings_form_context_add',
-    ),
-    '#ajax' => array(
-      'callback' => 'layout_ajax_form_open_dialog',
+  $form['context_wrapper']['context']['links'] = array(
+    '#theme' => 'layout_action_links',
+    'add' => array(
```

Comment:
Everywhere we have this pattern, could we write it this way instead?

```
  $form['context_wrapper']['context']['links'] = array(
    '#theme' => 'layout_action_links',
  );
  $form['context_wrapper']['context']['links']['add'] => array(
```

That would line up with the way we normally deal with nested elements.

Reply:
Please check the PR again (I just had the same thought, and made this update)